src/iop/atrous.c

```diff
@@ -251,90 +238,75 @@ static inline __m128 weight_sse2(const __m128 *c1, const __m128 *c2, const float
-  float *pdetail = detail + (size_t)4 * j * width;                                                           \
-  float *pcoarse = out + (size_t)4 * j * width;
-#endif
-
 #define SUM_PIXEL_PROLOGUE                                                                                   \
   float sum[4] = { 0.0f, 0.0f, 0.0f, 0.0f };                                                                 \
-  float wgt[4] = { 0.0f, 0.0f, 0.0f, 0.0f };
+  float wgt[4] = { 0.0f, 0.0f, 0.0f, 0.0f };                                                                 \
+  const size_t inc = 4 * (j * width + i);                                                                    \
+  const float *px = ((float *)in) + inc;                                                                     \
+  float *px2;                                                                                                \
+  float *pdetail = detail + inc;                                                                             \
+  float *pcoarse = out + inc;
 
 #if defined(__SSE2__)
 #define SUM_PIXEL_PROLOGUE_SSE                                                                               \
   __m128 sum = _mm_setzero_ps();                                                                             \
-  __m128 wgt = _mm_setzero_ps();
+  __m128 wgt = _mm_setzero_ps();                                                                             \
+  const size_t inc = 4 * (j * width + i);                                                                    \
+  const __m128 px = _mm_load_ps(in + inc);                                                                   \
+  float *pdetail = detail + inc;                                                                             \
+  float *pcoarse = out + inc;
 #endif
 
+
 #define SUM_PIXEL_EPILOGUE                                                                                   \
   for(int c = 0; c < 4; c++) sum[c] /= wgt[c];                                                               \
-                                                                                                             \
   for(int c = 0; c < 4; c++) pdetail[c] = (px[c] - sum[c]);                                                  \
-  for(int c = 0; c < 4; c++) pcoarse[c] = sum[c];                                                            \
-  px += 4;                                                                                                   \
-  pdetail += 4;                                                                                              \
-  pcoarse += 4;
+  for(int c = 0; c < 4; c++) pcoarse[c] = sum[c];
 
 #if defined(__SSE2__)
 #define SUM_PIXEL_EPILOGUE_SSE                                                                               \
-  sum = _mm_mul_ps(sum, _mm_rcp_ps(wgt));                                                                    \
```

Comment:
since sum is sse _m128 I think the we need to keep this line.

Reply:
this is fully equivalent for the compiler but only more readable